sample/sample.go

```diff
@@ -18,9 +20,38 @@ type Sampler interface {
 
 // SamplerFactory is used to create new samplers with common (injected) resources
 type SamplerFactory struct {
-	Config  config.Config   `inject:""`
-	Logger  logger.Logger   `inject:""`
-	Metrics metrics.Metrics `inject:"genericMetrics"`
+	Config        config.Config   `inject:""`
+	Logger        logger.Logger   `inject:""`
+	Metrics       metrics.Metrics `inject:"genericMetrics"`
+	Peers         peer.Peers      `inject:""`
+	peerCount     int
+	peerCountLock sync.RWMutex
+}
+
+func (s *SamplerFactory) GetClusterSize(useClusterSize bool) int {
+	if useClusterSize {
+		s.peerCountLock.RLock()
+		defer s.peerCountLock.RUnlock()
+		return s.peerCount
+	}
+	return 1
+}
+
+func (s *SamplerFactory) Start() error {
```

Comment:
Where does this end up getting invoked?

Reply:
Automatically as part of the injection of SamplerFactory from `app.go`.